### ISSUE TITLE
RNNCell error: no method matching cat(Operation, ...)

### ISSUE THREAD

**Issue by karenl7** [malmaud/TensorFlow.jl#11]
It seems like RNNCell may have gotten broken by the recent `Tensor` vs `Operation` changes. Defining
```julia
Base.cat(::Type{Operation}, dim, values...) = concat(dim-1, values)
```
doesn't seem to fix the issue, i.e. I'm trying to do something like
```julia
my_rnn_cell = nn.rnn_cell.BasicRNNCell(3)
my_rnn_cell(Variable(ones(10, 5)), Variable(ones(10, 3)))
```
Thanks for your help!

**Comment by karenl7** [comment]
Similarly, I think a similar issue is happening here:

https://github.com/malmaud/TensorFlow.jl/blob/31d7a1223e95459be73c0b31c13da085eda80395/src/rnn_cell.jl#L22

```julia
my_rnn_cell = nn.rnn_cell.BasicRNNCell(3)
nn.rnn_cell.zero_state(my_rnn_cell, 5, Float32)
```

produces the error

```julia
LoadError: MethodError: no method matching Array{TensorFlow.Operation,N}(::Type{Float32}, ::Tuple{Int64,Int64})
Closest candidates are:
  Array{TensorFlow.Operation,N}{T}(::Any) at sysimg.jl:53
while loading In[17], in expression starting on line 1

 in zero_state(::TensorFlow.nn.rnn_cell.BasicRNNCell, ::Int64, ::Type{T}) at /home/schmrlng/.julia/v0.5/TensorFlow/src/rnn_cell.jl:22
```

**Comment by malmaud** [comment]
You're right; the rnn code has to be updated. I'll do it soon and thanks for the report.

**Comment by malmaud** [comment]
Closed in 819dd49773a9cd38a142e84b68f913fb4bb79c68
